components/match2/commons/match_group_input.lua

```diff
@@ -583,6 +647,162 @@ function MatchGroupInput.readMvp(match)
 	return {players = parsedPlayers, points = mvppoints}
 end
 
+---@param match table
+---@param opponentIndex integer
+---@param opponentInput table
+---@return {players: table[], substitutions: string?}
+function MatchGroupInput.extractManualPlayersInput(match, opponentIndex, opponentInput)
+	--players from manual input as `opponnetX_pY`
+	local players = Array.mapIndexes(function(playerIndex)
+		return Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex])
+	end)
+	local manualInput = {players = players, substitutions = opponentInput.substitutes, timestamp = match.timestamp}
+
+	--players from manual input in `opponent.players`
+	local playersData = Json.parseIfString(opponentInput.players)
+
+	if not playersData then return manualInput end
+
+	for _, playerName, playerPrefix in Table.iter.pairsByPrefix(playersData, 'p') do
+		table.insert(players, {
+			pageName = playerName,
+			displayName = playersData[playerPrefix .. 'dn'],
+			flag = playersData[playerPrefix .. 'flag'],
+			faction = playersData[playerPrefix .. 'faction'] or playersData[playerPrefix .. 'race'],
+		})
+	end
+
+	return manualInput
+end
+
+---reads the players of a team from input and wiki variables
+---@param opponent standardOpponent
+---@param manualPlayersInput {players: table[], substitutions: string?}
+---@param options MatchGroupInputReadPlayersOfTeamOptions
+---@return standardPlayer[]
+---@return table
+function MatchGroupInput.readPlayersOfTeamNew(opponent, manualPlayersInput, options)
+	---@type string
+	local teamName = opponent.name
+
+	local players = {}
+	local playersIndex = 0
+
+	local insertIntoPlayers = function(player)
+		if type(player) ~= 'table' or Logic.isEmpty(player) or Logic.isEmpty(player.name or player.pageName) then
+			return
+		end
+
+		local pageName = Logic.nilIfEmpty(player.name) or player.pageName
+		pageName = options.resolveRedirect and mw.ext.TeamLiquidIntegration.resolve_redirect(pageName) or pageName
+		pageName = options.applyUnderScores and pageName:gsub(' ', '_') or pageName
+
+		playersIndex = playersIndex + 1
+		players[pageName] = Table.merge(players[pageName] or {}, {
+			pageName = pageName,
+			flage = Flags.CountryName(player.flag),
+			displayName = Logic.nilIfEmpty(player.displayname) or player.displayName,
+			faction = player.faction and Faction.read(player.faction) or nil,
+			index = playersIndex,
+		})
+	end
+
+	local playerIndex = 1
+	local varPrefix = teamName .. '_p' .. playerIndex
+	local name = Variables.varDefault(varPrefix)
+	-- if we do not find a player for the teamName try to find them for the teamName with underscores
+	if not name then
+		teamName = teamName:gsub(' ', '_')
+		varPrefix = teamName .. '_p' .. playerIndex
+	end
+
+	while name do
+		if options.maxNumPlayers and (playersIndex >= options.maxNumPlayers) then break end
```

Comment:
```suggestion
		if options.maxNumPlayers and (playersIndex >= options.maxNumPlayers) then break end
```
If there's a lot of not present in match (more than maxNumPlayayers), wouldn't this fail?

Reply:
tbh it is a straight copy from the existing code^^

technically yes it has its limits ...

i am open to kicking the manNumPlayers entirely fwiw
for sc/sc2/stormgate i for sure will not set it

keep in mind though that on most wikis TeamCard has a max number of players it sets the wiki vars for